docs/04-api-reference/front-commerce-core/ServerEvent.mdx

```diff
@@ -0,0 +1,45 @@
+---
+title: "ServerEvent"
+sidebar_position: 8
+description:
+  "The ServerEvent class define ServerEvent format used in Front-Commerce to be
+  published to the server event worker."
+---
+
+<p>{frontMatter.description}</p>
+
+## Usage:
+
+```ts
+import { ServerEvent } from "@front-commerce/core/events";
+
+// We define a static event Name for our event
+const MyEventName = "MyEvent";
+
+// Definition of the payload
+type MyEventPayload = {
+  sku: string;
+  price: boolean;
+};
+
+export default class MyEvent extends ServerEvent<MyEventPayload> {
+  static event_type = MyEventName;
+
+  constructor(sku: string, price: number, shopId: string) {
+    super(MyEventName, new Date(), { sku, price }, { shopId });
+  }
+}
+```
+
+When implementing this class, you must define the variable you want to
+manipulate in the constructor, then call `super` (which is ServerEvent
+constructor) with parameters formatted like the following.
```

Comment:
>  you must define the variable you want to manipulate in the constructor

**issue:** I'm not sure what this means. What variable are you talking about? Why does your example doesn't feature any variable in said constructor if it's necessary?

Reply:
This is kind of useless i think, i was thinking about explaining you can do stuff before calling super (like computing a new variable to pass to the event).

Reply:
removed